utilities/rpp-unittests/OCL/BatchPD_ocl.cpp

```diff
@@ -0,0 +1,729 @@
+    Rpp8u threshold[images];
+    Rpp32f threshold1[images];
+    Rpp32u nonmaxKernelSize[images];
+    for (i = 0; i < images; i++)
+    {
+        numOfPixels[i] = 4;
+        gaussianKernelSize[i] = 7;
+        kValue[i] = 1;
+        threshold[i] = 15;
+        threshold1[i] = 10;
+        nonmaxKernelSize[i] = 5;
+    }
+    Rpp32u sobelType[images];
+    for (i = 0; i < images; i++)
+    {
+        sobelType[i] = 1;
+    }
+    Rpp8u min[images];
+    Rpp8u max[images];
+    for (i = 0; i < images; i++)
+    {
+        min[i] = 10;
+        max[i] = 30;
+    }
+    const int size_perspective = images * 9;
+    Rpp32f perspective[64 * 9];
+    for (i = 0; i < images; i++)
+    {
+        perspective[0 + i * 9] = 1;
+        perspective[1 + i * 9] = 0.5;
+        perspective[2 + i * 9] = 0;
+        perspective[3 + i * 9] = 0.5;
+        perspective[4 + i * 9] = 1;
+        perspective[5 + i * 9] = 0;
+        perspective[6 + i * 9] = 1;
+        perspective[7 + i * 9] = 0;
+        perspective[8 + i * 9] = 0.5;
+    }
+    Rpp32f percentage[images];
+    for (i = 0; i < images; i++)
+    {
+        percentage[i] = 100;
+    }
+
+    cl_mem d_input, d_input_second, d_output;
+    cl_platform_id platform_id;
+    cl_device_id device_id;
+    cl_context theContext;
+    cl_command_queue theQueue;
+    cl_int err;
+    err = clGetPlatformIDs(1, &platform_id, NULL);
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
+    theContext = clCreateContext(0, 1, &device_id, NULL, NULL, &err);
+    theQueue = clCreateCommandQueue(theContext, device_id, 0, &err);
+    d_input = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_input_second = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_output = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_input, CL_TRUE, 0, ioBufferSize * sizeof(Rpp8u), input, 0, NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_input_second, CL_TRUE, 0, ioBufferSize * sizeof(Rpp8u), input_second, 0, NULL, NULL);
+    rppHandle_t handle;
+
+    rppCreateWithStreamAndBatchSize(&handle, theQueue, noOfImages);
+
+    clock_t start, end;
+    double cpu_time_used;
+    start = clock();
+    string test_case_name;
+    switch (test_case)
+    {
+    case 0:
+        test_case_name = "contrast";
+        rppi_contrast_u8_pkd3_batchPD_gpu(d_input, srcSize, maxSize, d_output, newMin, newMax, noOfImages, handle);
+        break;
+    case 1:
+        test_case_name = "jitter";
+        rppi_jitter_u8_pkd3_batchPD_gpu(d_input, srcSize, maxSize, d_output, kernelSize, noOfImages, handle);
+        break;
+    case 2:
+        test_case_name = "blur";
+        rppi_blur_u8_pkd3_batchPD_gpu(d_input, srcSize, maxSize, d_output, kernelSize, noOfImages, handle);
+        break;
+    case 3:
+        test_case_name = "brightness";
```

Comment:
Codacy found an issue: [Variable 'test_case_name' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)